### ISSUE TITLE
Error in gensim prepare when corpus is passed as a matrix

### ISSUE THREAD

**Issue by janrito** [bmabey/pyLDAvis#53]
A test is present in the `_extract_data` function in `gensim.py`, presumably so that the corpus can be passed as a sparse matrix:

``` .python
if not gensim.matutils.ismatrix(corpus):
      corpus_csc = gensim.matutils.corpus2csc(corpus)
   else:
      corpus_csc = corpus
```

Later, however, the length of the corpus is tested:

``` .python
   assert doc_lengths.shape[0] == len(corpus), 'Document lengths and corpus have different sizes {} != {}'.format(doc_lengths.shape[0], len(corpus))
```

When corpus is a sparse matrix, `len(corpus)` will raise an exception because the length of a sparse matrix is ambiguous.

**Comment by viig99** [comment]
Facing the same issue.

**Comment by bmabey** [comment]
I'll accept a PR that adds a conditional for this case.

**Comment by alexsalles** [comment]
I believe that the issue was solved in #82, in this [commit](https://github.com/bmabey/pyLDAvis/pull/82/commits/91f8bf078633784610d394978acae1f42872ba9c)

**Comment by bmabey** [comment]
Try the latest version, v2.1.1, to see if this is fixed for you.
